### ISSUE TITLE
How do I get started?

### ISSUE THREAD

**Issue by Yinlin124** [nirtamir2/obsidian-slidev#1]
I'm unable to install this Obsidian plugin either through community plugins or via BRAT. I'm really looking forward to using this plugin. Could you please guide me on how to get started?

**Comment by nirtamir2** [comment]
Hi!
This plugin is still a work in progress.
I can notify you with detailed instructions when it will be ready :)
BTW - there is a prerequisite of having node.js installed on your computer.
Is this prerequisite work for you?
Thank you for interest ❤️

**Comment by Yinlin124** [comment]
Yes, I have got Node.js ready. I see the commands in the current README are for Mac. Does this plugin support Windows environment as well?

**Comment by Yinlin124** [comment]
> Hi! Thank you for interest ❤️ This plugin is still a work in progress. I can notify you with detailed instructions when it will be ready :) BTW - there is a prerequisite of having node.js installed on your computer. Is this prerequisite work for you?

Yes, I have got Node.js ready. I see the commands in the current README are for Mac. Does this plugin support Windows environment as well?

**Comment by nirtamir2** [comment]
I haven't checked it on Windows. I configure it to run a command specific to my env to load node.js. We can also configure it to run on Windows (or I will supply a setting for the initial load script because somehow my node.js did not load natively in child_proccess).

**Comment by Yinlin124** [comment]
> I haven't checked it on Windows. I configure it to run a command specific to my env to load node.js. We can also configure it to run on Windows (or I will supply a setting for the initial load script because somehow my node.js did not load natively in child_proccess).

Well, I can't wait to experience this plugin, because slidev is usually edited markdown content in vscode, but that doesn't fit my workflow. I look forward to your development.

**Comment by nirtamir2** [comment]
I update the readme with development instructions. You can now clone the repo, install deps and run the dev command. It will help me to get early feedback

**Comment by Yinlin124** [comment]
![image](https://github.com/nirtamir2/obsidian-slidev/assets/91457024/642f66cd-1725-4559-b160-be863a8be021)

I successfully compiled this project, the plugin seems pretty good so far, and the experience is better than vscode. 

But, did you use this plugin by creating a new slidev project folder directly under the obsidian vault? 

Because obsidian cannot open files outside the vault, this undoubtedly increases the complexity of the vault.

**Comment by Yinlin124** [comment]
![image](https://github.com/nirtamir2/obsidian-slidev/assets/91457024/56fdb8b6-8b48-4300-8eec-515a0a312959)

By the way, there are also some minor garbled text bugs，but it doesn't affect anything.

**Comment by nirtamir2** [comment]
Hi @Yinlin124 
I'm really happy to see your feedback!

https://github.com/nirtamir2/obsidian-slidev/issues/1#issuecomment-1675748081 Seems like it's working in windows too 🥳

https://github.com/nirtamir2/obsidian-slidev/issues/1#issuecomment-1675749399 - I believe this causes by the setting that does the `source ~/.profile` command. If you remove it - it should not error in windows (TODO: I should make the default empty when I detect windows os)

> But, did you use this plugin by creating a new slidev project folder directly under the obsidian vault?

This is tricky. Currently - I add the template in the repo (this is why it did not work with BRAT or I haven't publish it to obsidian). I plan to add instructions for the users / make an action that will init slidev to a place configured in settings. I believe it should work. Meaning that the plugin will check for the template in the given location and prompt the user to create the template manually / with a button. WDYT?

**Comment by Yinlin124** [comment]
> This is tricky. Currently - I add the template in the repo (this is why it did not work with BRAT or I haven't publish it to obsidian). I plan to add instructions for the users / make an action that will init slidev to a place configured in settings. I believe it should work. Meaning that the plugin will check for the template in the given location and prompt the user to create the template manually / with a button. WDYT?

![image](https://github.com/nirtamir2/obsidian-slidev/assets/91457024/bf3142e4-e9c0-4a71-8ead-d101bfc8275b)


Hi! nirtamir2
I think this method is feasible.

In addition, I tried another approach which also seems to work - creating symbolic links( **.symlink** i donot how to translate) in the vault for markdown files edited for slidev projects in other folders also takes effect.

I hope this can give you some inspiration.

**Comment by nirtamir2** [comment]
I just publish v0.0.5 with support for a custom slidev template. Let me know if it works for you

**Comment by Yinlin124** [comment]
Hi! nirtamir2.
It seem didnot work on Windows.

Bugs:
The first time I opened the plugin without setting the slidev-template path, it said the path was already valid even though it didn't exist yet. There might be some issue with the path validation. 
![image](https://github.com/nirtamir2/obsidian-slidev/assets/91457024/2e22d87c-d6e4-4243-966b-37d5ad6c5df9)

After I properly configured the slidev-template path (I put it under ..\.obsidian\plugins\slidev\slidev-template, and node.js environment variable is set up on my computer)

It didnot work.
![image](https://github.com/nirtamir2/obsidian-slidev/assets/91457024/7041c29a-67ad-4280-a90f-e6031959f822)

And log is "child process exited with code 0 and signal null child process exited with code 0" or "empty"

![image](https://github.com/nirtamir2/obsidian-slidev/assets/91457024/08514d11-b7a9-4b62-a48a-051b908f00c7)

I have absolutely no idea where the problem. Hope I can provide some help. Looking forward to your continued development.

**Comment by nirtamir2** [comment]
Did you run `npm install` in the template directory?
In order to check for valid path - I check if I can run slidev in this path. Is it installed globally?

**Comment by Yinlin124** [comment]
The path detection was indeed a global installation issue. After I uninstalled globally, the path detection became valid.



> Did you run `npm install` in the template directory?

I have already npm installed. When I start it myself using bash, the plugin works. But it does not work when launched through commands within Obsidian.

**Comment by Yinlin124** [comment]
![image](https://github.com/nirtamir2/obsidian-slidev/assets/91457024/bb78cf6c-69bf-4418-8c71-f2cd55a94ce0)


When I start it myself using bash, the plugin works. But it does not work when launched through commands within Obsidian.
